### ISSUE TITLE
Safari 12 compatibility issues

### ISSUE THREAD

**Issue by PTDennis** [el1t/uBlock-Safari#140]
The updated Safari 12 detects uBlock Origin as slowing down the web pages loading times. 

When will be available a fix for this?

**Comment by georgeperez** [comment]
It is very likely that this project has been abandoned. Please look at #130.

**Comment by MikeWilkie** [comment]
Here's a "fix":

1. Download [release 1.15.4](https://github.com/el1t/uBlock-Safari/releases/tag/1.15.4)
2. Follow [these instructions to install manually](https://georgegarside.com/blog/macos/install-any-safari-extension-macos-mojave/)

Why 1.15.4 and not 1.16.0 you ask? Two reasons:

1. You will not get the error from Safari
2. I don't know about you, but I've never been able to lock changes with the "lock" in version 1.16.0. I'll click the button and...nothing. Version 1.15.4 does not have this issue.

Yes, version 1.15.4 is missing several upstream changes, but considering it "works" in all other ways, this is what I'm sticking with.

**Comment by zonetuke** [comment]
Bummer. I prefer uBlock Origin over AdGuard.

**Comment by LesterJitsi** [comment]
I'll stick with Safari 11 and 10.11.6, but fwiw here's a workaround maybe you already knew (2nd comment):
https://www.macrumors.com/2018/09/17/apple-releases-safari-12-macos-high-sierra

**Comment by jnoxon** [comment]
> I'll stick with Safari 11 and 10.11.6, but fwiw here's a workaround maybe you already knew (2nd comment):
> https://www.macrumors.com/2018/09/17/apple-releases-safari-12-macos-high-sierra

Works great, thanks!

**Comment by kurabi** [comment]
@LesterJitsi @jnoxon It won't unzip for me when I change the extension to a zip. Any clues?

**Comment by kurabi** [comment]
Nevermind, it unzipped correctly with `xar -xf uBlock0.safariextz`

**Comment by ghost** [comment]
I just first up want to **thank** @el1t l1t for his work, I enjoyed u block origin and would like to use it in safari in the future as well.
In the past he managed to get reasonable updates and keep the extension working well, **nice work**! 👍

**Comment by scana** [comment]
To sum things up, without having to rely on a comment from external website:
- download the extension from here: https://github.com/el1t/uBlock-Safari/releases
- unpack it with `xar -xf uBlock0.safariextz`
- Safari -> Develop menu -> Show Extension Builder -> "+" button in lower-left corner -> Add extension -> point to unpacked `uBlock0.safariextension` folder
-- if you don't have develop menu available: Safari -> Preferences -> Advanced tab -> "Show Develop menu in menu bar"
- click "Run" in upper-right corner

**Comment by sampl3x** [comment]
@scana When you close and open safari again Ublock is disabled again, so it not a workable solution.

**Comment by mustii82** [comment]
Just go to extensions and enable the extension again safari will warn you that the add-on can slow the browser down but after that it will work

**Comment by sampl3x** [comment]
Yeah i know now but you have to do it overtime when you have quit Safari.

**Comment by anktkr** [comment]
https://github.com/el1t/uBlock-Safari/issues/140#issuecomment-423473861
> To sum things up, without having to rely on a comment from external website:
> 
> * download the extension from here: https://github.com/el1t/uBlock-Safari/releases
> * unpack it with `xar -xf uBlock0.safariextz`
> * Safari -> Develop menu -> Show Extension Builder -> "+" button in lower-left corner -> Add extension -> point to unpacked `uBlock0.safariextension` folder
>   -- if you don't have develop menu available: Safari -> Preferences -> Advanced tab -> "Show Develop menu in menu bar"
> * click "Run" in upper-right corner

Find an AppleScript to do these steps automatically here: 
https://georgegarside.com/blog/macos/install-any-safari-extension-macos-mojave/
